### PR TITLE
Adapt Ansible playbooks to run functional test

### PR DESCRIPTION
**Pull request by Paullgk**
This PR modifies Ansible playbooks in order to run functional test for each machine (hypervisor and observer) from an the playbook test_run_cukinia.yaml.

### CODE REVIEW

File: playbooks/test_deploy_cukinia_tests.yaml [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
 - name: Deploy Cukinia's tests
-  hosts: cluster_machines
+  hosts: hypervisors
   tasks:
     - name: Copy Cukinia's tests
       copy:
         src: ../src/cukinia-tests/cukinia
         dest: /etc/
+    - name: Copy cukinia.conf
+      template:
```
Comment (dupremathieu) [inline playbooks/test_deploy_cukinia_tests.yaml:13]: cukinia-hypervisor.conf.j2 is not a template but regular file. Use copy instead.

File: playbooks/test_deploy_cukinia_tests.yaml [hunk 2]
```diff
@@ -36,5 +40,41 @@
...
+- name: Deploy Cukinia's tests
+  hosts: observers
+  tasks:
+    - name: Copy Cukinia's tests
+      copy:
+        src: ../src/cukinia-tests/cukinia
+        dest: /etc/
+    - name: Copy cukinia.conf
+      template:
+        src: ../templates/cukinia-observer.conf.j2
```
Comment (dupremathieu) [inline playbooks/test_deploy_cukinia_tests.yaml:52]: cukinia-observer.conf.j2 is not a template but regular file. Use copy instead.

File: playbooks/test_deploy_cukinia_tests.yaml [hunk 3]
```diff
@@ -36,5 +40,41 @@
...
+        dest: /etc/cukinia/cukinia.conf
+    - name: Create /usr/share/cukinia/includes
```
Comment (dupremathieu) [inline playbooks/test_deploy_cukinia_tests.yaml:54]: Cukinia includes should be copy in hypervisors too.

File: playbooks/test_deploy_cukinia_tests.yaml [hunk 4]
```diff
@@ -36,5 +40,41 @@
...
+      file:
+        path: /usr/share/cukinia/includes
+        state: directory
+        owner: root
+        group: root
+        mode: 0755
+    - name: Copy Cukinia's includes
+      copy:
+        src: ../src/cukinia-tests/includes/kernel_config_functions
```
Comment (dupremathieu) [inline playbooks/test_deploy_cukinia_tests.yaml:63]: kernel_config_functions should be copy in hypervisors too.

File: templates/cukinia-hypervisor.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Copyright (C) 2020, RTE (http://www.rte-france.com)
```
Comment (dupremathieu) [inline templates/cukinia-hypervisor.conf.j2:1]: 2023

File: templates/cukinia-hypervisor.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+# SPDX-License-Identifier: Apache-2.0
```
Comment (dupremathieu) [inline templates/cukinia-hypervisor.conf.j2:2]: I will move this file inside the cukinia-tests submodule

File: templates/cukinia-observer.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2020, RTE (http://www.rte-france.com)
```
Comment (dupremathieu) [inline templates/cukinia-observer.conf.j2:1]: 2023

File: templates/cukinia-observer.conf.j2 [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# SPDX-License-Identifier: Apache-2.0
```
Comment (dupremathieu) [inline templates/cukinia-observer.conf.j2:2]: I will move this file inside the cukinia-tests submodule.

**Review by dupremathieu — CHANGES_REQUESTED** [review]

File: playbooks/test_deploy_cukinia_tests.yaml (hunk 3 above)
Comment (Paullgk) [reply playbooks/test_deploy_cukinia_tests.yaml:54]: They already are

File: playbooks/test_deploy_cukinia_tests.yaml (hunk 4 above)
Comment (Paullgk) [reply playbooks/test_deploy_cukinia_tests.yaml:63]: They already are

File: templates/cukinia-hypervisor.conf.j2 (hunk 6 above)
Comment (Paullgk) [reply templates/cukinia-hypervisor.conf.j2:2]: Ack, see the PR on cukinia-tests repo

File: templates/cukinia-observer.conf.j2 (hunk 7 above)
Comment (Paullgk) [reply templates/cukinia-observer.conf.j2:1]: Ack, see the PR on cukinia-tests repo

File: templates/cukinia-observer.conf.j2 (hunk 8 above)
Comment (Paullgk) [reply templates/cukinia-observer.conf.j2:2]: Ack, see the PR on cukinia-tests repo

File: playbooks/test_deploy_cukinia_tests.yaml [hunk 9]
```diff
@@ -3,12 +3,16 @@
 
 ---
 - name: Deploy Cukinia's tests
-  hosts: cluster_machines
+  hosts: hypervisors
```
Comment (dupremathieu) [inline playbooks/test_deploy_cukinia_tests.yaml:6]: Maybe you can merge the common content:
```yml
---
- name: Deploy Cukinia's tests
  hosts: cluster_machines
  tasks:
    - name: Copy Cukinia's tests
      copy:
        src: ../src/cukinia-tests/cukinia
        dest: /etc/
    - name: Create /usr/share/cukinia/includes
      file:
        path: /usr/share/cukinia/includes
        state: directory
        owner: root
        group: root
        mode: 0755
    - name: Copy Cukinia's includes
      copy:
        src: ../src/cukinia-tests/includes/kernel_config_functions
        dest: /usr/share/cukinia/includes/kernel_config_functions
    - name: Create /usr/share/testdata
      file:
        path: /usr/share/testdata
        state: directory
        owner: root
        group: root
        mode: 0755
    - name: Copy vm.xml
      copy:
        src: ../src/debian/vm_manager/vm_manager/testdata/vm.xml
        dest: /usr/share/testdata
    - name: Copy wrong_vm_config.xml
      copy:
        src: ../src/debian/vm_manager/vm_manager/testdata/wrong_vm_config.xml
        dest: /usr/share/testdata


- name: Deploy Cukinia's main configuration on observers
  hosts: observers
  tasks:
    - name: Copy cukinia.conf
      copy:
        src: ../src/cukinia-tests/cukinia-observer.conf
        dest: /etc/cukinia/cukinia.conf

- name: Deploy Cukinia's main configuration on hypervisor
  hosts: hypervisor
  tasks:
    - name: Copy cukinia.conf
      copy:
        src: ../copy/cukinia-tests/cukinia-hypervisor.conf
        dest: /etc/cukinia/cukinia.conf
```

File: playbooks/test_run_cukinia.yaml [hunk 10]
```diff
@@ -1,16 +1,16 @@
 # Copyright (C) 2022, RTE (http://www.rte-france.com)
 # SPDX-License-Identifier: Apache-2.0
 
-# Ansible playbook that runs Cukinia tests and export the result in CSV.
+# Ansible playbook that runs Cukinia functional tests and export the result in
+# CSV.
 
 ---
-- hosts: cluster_machines
+- hosts: observers
```
Comment (dupremathieu) [inline playbooks/test_run_cukinia.yaml:8]: cluster_machines

File: playbooks/test_run_cukinia.yaml [hunk 11]
```diff
@@ -21,11 +21,32 @@
...
-          {{ cukinia_configuration }} || true
+          -o {{ tmp_cukinia_dir.path }}/{{ tests_result_name }} || true
     - name: Fetch result
       fetch:
         src: "{{ tmp_cukinia_dir.path }}/{{ tests_result_name }}"
         dest: "{{ cukinia_test_prefix }}/{{ tests_result_name }}"
         flat: yes
-  run_once: true
+
+- hosts: hypervisors
```
Comment (dupremathieu) [inline playbooks/test_run_cukinia.yaml:31]: Remove this part

File: playbooks/test_run_cukinia-sec.yaml [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+          MACHINENAME={{ cukinia_namespace | default(inventory_hostname) }}
+          cukinia -f {{ tests_format }}
+          -o {{ tmp_cukinia_dir.path }}/{{ tests_result_name }}
+          {{ cukinia_configuration }} || true
+    - name: Fetch result
+      fetch:
+        src: "{{ tmp_cukinia_dir.path }}/{{ tests_result_name }}"
+        dest: "{{ cukinia_test_prefix }}/{{ tests_result_name }}"
+        flat: yes
+  run_once: true
```
Comment (dupremathieu) [inline playbooks/test_run_cukinia-sec.yaml:31]: Why run this only in one machine?

**Review by dupremathieu — CHANGES_REQUESTED** [review]

File: playbooks/test_run_cukinia-sec.yaml (hunk 12 above)
Comment (Paullgk) [reply playbooks/test_run_cukinia-sec.yaml:31]: It was already here in previous version of test_run_cukinia. I removed it

**Pull request closed without merging**